src/main/scala/scalaz/DistributedModule.scala

```diff
@@ -9,24 +9,44 @@ trait Member
 sealed trait Membership
 
 object Membership {
-  final case class Join(member: Member)        extends Membership
-  final case class Leave(member: Member)       extends Membership
+
+  final case class Join(member: Member) extends Membership
+
+  final case class Leave(member: Member) extends Membership
+
   final case class Unreachable(member: Member) extends Membership
+
 }
 
 sealed trait DistributedError
 
 case class MetadataID(v: String)
 
 trait DistributedModule {
-
   type F[A] = IO[DistributedError, A]
 
   type Type[A]
 
-  type Lens[A, B]
-  type Prism[A, B]
-  type Traversal[A, B]
+  type Path[A, B] = PathElem[A, B]
+
+  sealed trait PathElem[A, B]
+
+  private object PathElem {
+
+    case class Key[K: Type, V: Type](v: K) extends PathElem[Map[K, V], V]
```

Comment:
Please also finalize these.

Reply:
what do mean by that?

Reply:
Just make them `final`.

Reply:
understood. done

Reply:
sorry but this will cause: Error: The outer reference in this type test cannot be checked at run time.